src/mobile-ui-react/ModalEntryFormDialog.tsx

```diff
@@ -169,19 +170,27 @@ export function ModalDialog(props: ModalDialogProps) {
     }, 0);
   }, []);
 
-  const handleOK = async () => {
+  const handleOK = React.useCallback(async () => {
+    if (waitingForOK)
+      return false;
+    setWaitingForOK(true);
     if (await onOK()) {
       setFadedOut(true);
       return true;
     }
+    setWaitingForOK(false);
     return false;
-  };
+  }, [onOK, waitingForOK]);
 
-  const handleCancel = () => {
+  const handleCancel = React.useCallback(() => {
+    if (waitingForOK)
+      return;
```

Comment:
Do we need to guard against multiple clicks of the Cancel button too?

Reply:
That's not what this does. It prevents the dialog from being canceled while we are waiting on the result of the OK press. Both OK and Cancel are visually disabled when the user hits OK, and so we don't want Cancel to work anyway.

Reply:
Right, I was wondering if we could get into a situation where Cancel was pressed and the buttons weren't disabled, so it could get pressed again (or OK could be pressed). But I see that the cancel handler isn't async, so it shouldn't be an issue.

Reply:
The onCancel handler isn't async, but the dialog does still have to fade out after it is pressed. I'll tweak the code to make sure they can't be pressed again after that.

Reply:
Instead of `waitingForOK`, it's now `waitingToClose`, and it gets set for both OK and Cancel.